hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -180,6 +181,31 @@ public final JobStatus getStatus() {
 
     protected abstract JobStatus getStatus0();
 
+    @Override
+    public final boolean isUserCancelled() {
+        if (isLightJob()) {
+            CompletableFuture<Void> f = getFuture();
+            if (!f.isDone()) {
+                throw new IllegalStateException("Job not finished");
+            }
+            if (!f.isCancelled()) {
+                return false;
+            }
+            try {
+                f.getNow(null);
```

Comment:
`getNow` should not throw `CompletionException` because there is `if (!f.isCancelled())` check before

Reply:
`if (!f.isCancelled())` branch is tested in a few tests with job failed with exception, for example `ExecutionLifecycleTest.when_pmsInitThrows_then_jobFails`.

Reply:
BTW, if there was a problem with client communication with Hazelcast node when waiting for the job result (e.g. connection reset, timeout), it would be perfect to throw that exception instead of returning false from `isUserCancelled`. But I think we do not care.